doc/samples/sample_notify_fifo.sh

```diff
@@ -24,9 +24,24 @@ FIFO=$1
 [[ -z $FIFO ]] && echo "A FIFO name must be specified" && exit 1
 
 LOG_FILE=/tmp/${FIFO##*/}.log
+if [[ -d /run ]]; then
+    PID_DIR=/run
+elif [[ -d /var/run ]]; then
+    PID_DIR=/var/run
+else
+    PID_DIR=/tmp
+fi
+PID_FILE=$PID_DIR/${FIFO##*/}.pid
+
+exiting()
+{
+	rm -f "$PID_FILE"
+	[[ -n $FD ]] && exec {FD}>&-
```

Comment:
I suggest remove
   `[[ -n $FD ]] && exec {FD}>&-`
since it is not needed.

Also add:
```
    [[ $CREATED_FIFO -eq 1 ]] && rm -f $FIFO

    flock -u $FD
```

Adding `flock -u $FD` significantly speeds up the `flock -e`. In my tests, without `flock -u` the `flock -e` took 9.78 seconds, with the `flock -u` it took 0.000008 seconds.

Moving the `rm -f $FIFO` from `stopping()` to `exiting()` ensures that the FIFO is always removed if we created it. Removing the FIFO before releasing the lock ensures that we don't remove the FIFO after the new instance of the script has checked for its existence (providing the lock is obtained before the FIFO is created - see below).